### PR TITLE
Add argument support to regex

### PR DESCRIPTION
**Pull request by gfs**
Adds options for IgnoreCase and MultiLine regexes using new Arguments field in Clause.

Required for DevSkim/ApplicationInspector.

### CODE REVIEW

File: OAT/Operations/RegexOperation.cs [hunk 1]
```diff
@@ -54,9 +54,20 @@ internal OperationResult RegexOperationDelegate(Clause clause, object? state1, o
...
+                }
+                if (clause.Arguments.Contains("m"))
+                {
+                    options |= RegexOptions.Multiline;
+                }
+
                 var built = string.Join("|", RegexList);
 
-                var regex = StringToRegex(built);
+                var regex = StringToRegex(built,options);
```
Comment (scovetta) [inline OAT/Operations/RegexOperation.cs:70]: Nit, formatting.

File: OAT/Operations/RegexOperation.cs [hunk 2]
```diff
@@ -103,22 +114,23 @@ internal OperationResult RegexOperationDelegate(Clause clause, object? state1, o
         /// Uses an internal cache.
         /// </summary>
         /// <param name="built">The regex to build</param>
+        /// <param name="regexOptions">The options to use.</param>
         /// <returns>The built Regex</returns>
-        public Regex? StringToRegex(string built)
+        public Regex? StringToRegex(string built, RegexOptions regexOptions)
         {
-            if (!RegexCache.ContainsKey(built))
+            if (!RegexCache.ContainsKey((built,regexOptions)))
```
Comment (scovetta) [inline OAT/Operations/RegexOperation.cs:121]: Nit, formatting.

File: OAT/Operations/RegexOperation.cs [hunk 3]
```diff
@@ -103,22 +114,23 @@ internal OperationResult RegexOperationDelegate(Clause clause, object? state1, o
...
             {
                 try
                 {
-                    RegexCache.TryAdd(built, new Regex(built, RegexOptions.Compiled));
+                    RegexCache.TryAdd((built,regexOptions), new Regex(built, regexOptions));
```
Comment (scovetta) [inline OAT/Operations/RegexOperation.cs:125]: Nit, formatting.

File: OAT/Operations/RegexOperation.cs [hunk 4]
```diff
@@ -103,22 +114,23 @@ internal OperationResult RegexOperationDelegate(Clause clause, object? state1, o
...
                 }
                 catch (ArgumentException)
                 {
                     Log.Warning("InvalidArgumentException when creating regex. Regex {0} is invalid and will be skipped.", built);
                     RegexCache.TryAdd(built, null);
                 }
             }
-            return RegexCache[built];
+            return RegexCache[(built,regexOptions)];
```
Comment (scovetta) [inline OAT/Operations/RegexOperation.cs:133]: Nit, formatting.

File: OAT/Operations/RegexOperation.cs (hunk 1 above)
Comment (gfs) [reply OAT/Operations/RegexOperation.cs]: Thanks for the catches.  Ran the formatter to resolve all.

**Pull request merged**